src/main/java/org/launchcode/liftoffrecipetracker/controllers/RecipeController.java

```diff
@@ -54,54 +59,73 @@ public String displayRecipes(@RequestParam(required = false) Integer categoryId,
+				model.addAttribute("recipes", user.getRecipes());
+			}
+
+		return "recipe/index";
 	}
 
-	@PostMapping("create")
-	public String processCreateRecipeForm(@ModelAttribute @Valid Recipe newRecipe,
-	                                      Errors errors, Model model,
-	                                      @RequestParam(required = false) List<Integer> categories, @RequestParam(required = false) List<Integer> beverages) {
-		if (errors.hasErrors()) {
+		@GetMapping("create")
+		public String displayCreateRecipeForm (Model model, HttpSession userSession){
 			model.addAttribute("title", "Create a Recipe");
 			model.addAttribute("categories", categoryRepository.findAll());
 			model.addAttribute("beverages", beverageRepository.findAll());
+			model.addAttribute("users", authenticationController.getUserFromSession(userSession));
+			model.addAttribute(new Recipe());
 			return "recipe/create";
-		} else if ((categories != null) && (beverages != null)) {
-			List<Category> categoryObjects = (List<Category>) categoryRepository.findAllById(categories);
-			newRecipe.addCategories(categoryObjects);
-			List<Beverage> beverageObjects = (List<Beverage>) beverageRepository.findAllById(beverages);
-			newRecipe.addBeverages(beverageObjects);
-		}else if (categories != null) {
+		}
+
+		@PostMapping("create")
+		public String processCreateRecipeForm (@ModelAttribute @Valid Recipe newRecipe,
+				Errors errors, Model model,
+				@RequestParam(required = false) List < Integer > categories,
+				@RequestParam(required = false) List < Integer > beverages, HttpSession userSession){
+
+
+			if (errors.hasErrors()) {
+				model.addAttribute("title", "Create a Recipe");
+				model.addAttribute("categories", categoryRepository.findAll());
+				model.addAttribute("beverages", beverageRepository.findAll());
+				model.addAttribute("users", authenticationController.getUserFromSession(userSession));
+				return "recipe/create";
+			} else if ((categories != null) && (beverages != null)) {
+				List<Category> categoryObjects = (List<Category>) categoryRepository.findAllById(categories);
+				newRecipe.addCategories(categoryObjects);
+				List<Beverage> beverageObjects = (List<Beverage>) beverageRepository.findAllById(beverages);
+				newRecipe.addBeverages(beverageObjects);
+			} else if (categories != null) {
 				List<Category> categoryObjects = (List<Category>) categoryRepository.findAllById(categories);
 				newRecipe.addCategories(categoryObjects);
-			} else if (beverages !=null){
-			List<Beverage> beverageObjects = (List<Beverage>) beverageRepository.findAllById(beverages);
-			newRecipe.addBeverages(beverageObjects);}
+			} else if (beverages != null) {
+				List<Beverage> beverageObjects = (List<Beverage>) beverageRepository.findAllById(beverages);
+				newRecipe.addBeverages(beverageObjects);
+			}
+			else if (userSession != null) {
+				User user = authenticationController.getUserFromSession(userSession);
+					newRecipe.setUser(user);
```

Comment:
I'm not sure if the issue with the "Created by:" switching when new users are logged in is an issue in the logic here or the logic in the display fields. I'm pretty sure it's in the display fields and I think we should start there before we go switching the logic here around. From what I can tell, you have it set up here the right way!